## Clone DeepSpeech v0.7.4 from GitHub

Download corresponding checkpoint also

`/content/model_checkpoints/ ` is where we'll be storing our model checkpoints


In [ ]:
!git clone --branch v0.7.4 https://github.com/mozilla/DeepSpeech

!wget https://github.com/mozilla/DeepSpeech/releases/download/v0.7.4/deepspeech-0.7.4-checkpoint.tar.gz
!tar -xvf deepspeech-0.7.4-checkpoint.tar.gz -C /content/model_checkpoints/ 

Cloning into 'DeepSpeech'...
remote: Enumerating objects: 38, done.
remote: Counting objects: 100% (38/38), done.
remote: Compressing objects: 100% (35/35), done.
remote: Total 23281 (delta 5), reused 14 (delta 0), pack-reused 23243
Receiving objects: 100% (23281/23281), 49.07 MiB | 27.84 MiB/s, done.
Resolving deltas: 100% (16015/16015), done.
Note: checking out 'fcd9563fcd8b47ee5719b24a9d7f0d9a4eaf372f'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

--2021-02-16 13:27:14--  https://github.com/mozilla/DeepSpeech/releases/download/v0.7.4/deepspeech-0.7.4-checkpoint.tar.gz
Resolving github.com (github.com)... 140.82.114.4
Connecting to

## Install DeepSpeech Dependencies

In [ ]:
%cd /content/
!sudo apt-get install python3-venv
!sudo apt-get install python3-dev
!pip install --upgrade pip
!sudo apt-get install sox
!sudo apt-get install sox libsox-fmt-mp3
!sudo apt install git
!pip install librosa==0.7.2
!sudo apt-get install pciutils
!lspci | grep -i nvidia

/content
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  python-pip-whl python3.6-venv
The following NEW packages will be installed:
  python-pip-whl python3-venv python3.6-venv
0 upgraded, 3 newly installed, 0 to remove and 17 not upgraded.
Need to get 1,660 kB of archives.
After this operation, 1,902 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 python-pip-whl all 9.0.1-2.3~ubuntu1.18.04.4 [1,653 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 python3.6-venv amd64 3.6.9-1~18.04ubuntu1.3 [6,180 B]
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 python3-venv amd64 3.6.7-1~18.04 [1,208 B]
Fetched 1,660 kB in 1s (2,156 kB/s)
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/s

In [ ]:
!wget https://github.com/git-lfs/git-lfs/releases/download/v2.11.0/git-lfs-linux-amd64-v2.11.0.tar.gz
!tar xvf /content/git-lfs-linux-amd64-v2.11.0.tar.gz -C /content
!sudo bash /content/install.sh
%cd /content/DeepSpeech
!git-lfs pull

--2021-02-16 13:27:55--  https://github.com/git-lfs/git-lfs/releases/download/v2.11.0/git-lfs-linux-amd64-v2.11.0.tar.gz
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github-releases.githubusercontent.com/13021798/fa85ce00-9147-11ea-9ec4-c204e7a4e6cd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210216%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210216T132755Z&X-Amz-Expires=300&X-Amz-Signature=e4b1be3f681c640ae294aae528bfbec5b18faf1c52dccc295b70653a86ca16a3&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=13021798&response-content-disposition=attachment%3B%20filename%3Dgit-lfs-linux-amd64-v2.11.0.tar.gz&response-content-type=application%2Foctet-stream [following]
--2021-02-16 13:27:55--  https://github-releases.githubusercontent.com/13021798/fa85ce00-9147-11ea-9ec4-c204e7a4e6cd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-

In [ ]:
!wget https://github.com/mozilla/DeepSpeech/releases/download/v0.7.4/ds_ctcdecoder-0.7.4-cp36-cp36m-manylinux1_x86_64.whl
!pip install /content/DeepSpeech/ds_ctcdecoder-0.7.4-cp36-cp36m-manylinux1_x86_64.whl

--2021-02-16 13:28:40--  https://github.com/mozilla/DeepSpeech/releases/download/v0.7.4/ds_ctcdecoder-0.7.4-cp36-cp36m-manylinux1_x86_64.whl
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github-releases.githubusercontent.com/60273704/743cbd00-b1c6-11ea-96f6-79d96377b886?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20210216%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210216T132840Z&X-Amz-Expires=300&X-Amz-Signature=aaa42843d89e5e3152e800f05445f8effbe0e7bc8f621e1d1128506b8ec7b309&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=60273704&response-content-disposition=attachment%3B%20filename%3Dds_ctcdecoder-0.7.4-cp36-cp36m-manylinux1_x86_64.whl&response-content-type=application%2Foctet-stream [following]
--2021-02-16 13:28:40--  https://github-releases.githubusercontent.com/60273704/743cbd00-b1c6-11ea-96f6-79d96377b8

In [ ]:
!pip3 install folium==0.2.1
!pip3 install --upgrade pip==20.0.2 wheel==0.34.2 setuptools==46.1.3
!pip3 install --upgrade --force-reinstall -e .

# Restart Colab after this cell

     |████████████████████████████████| 69 kB 6.7 MB/s 
  Created wheel for folium: filename=folium-0.2.1-py3-none-any.whl size=79978 sha256=97034a0691afea5801fb55a8ac3747f1d85701d784324ef0c73c989053405ae5
  Stored in directory: /root/.cache/pip/wheels/b9/9d/f7/42a080a41839b2db3c3572a0c81cc17a88a7f4ba890741c621
Successfully built folium
  Attempting uninstall: folium
    Found existing installation: folium 0.8.3
    Uninstalling folium-0.8.3:
      Successfully uninstalled folium-0.8.3
     |████████████████████████████████| 1.4 MB 20.0 MB/s 
     |████████████████████████████████| 582 kB 70.8 MB/s 
  Attempting uninstall: wheel
    Found existing installation: wheel 0.36.2
    Uninstalling wheel-0.36.2:
      Successfully uninstalled wheel-0.36.2
  Attempting uninstall: setuptools
    Found existing installation: setuptools 53.0.0
    Uninstalling setuptools-53.0.0:
      Successfully uninstalled setuptools-53.0.0
  Attempting uninstall: pip
    Found existing installation: pip 21.0.1

Obtaining file:///content/DeepSpeech
     |████████████████████████████████| 14.8 MB 132 kB/s 
     |████████████████████████████████| 49 kB 8.7 MB/s 
     |████████████████████████████████| 127 kB 86.2 MB/s 
     |████████████████████████████████| 287 kB 70.8 MB/s 
     |████████████████████████████████| 9.5 MB 61.0 MB/s 
     |████████████████████████████████| 61 kB 10.9 MB/s 
     |████████████████████████████████| 3.7 MB 22.6 MB/s 
     |████████████████████████████████| 20.2 MB 1.3 MB/s 
     |████████████████████████████████| 183 kB 78.4 MB/s 
     |████████████████████████████████| 1.9 MB 64.7 MB/s 
     |████████████████████████████████| 110.5 MB 76 kB/s 
     |████████████████████████████████| 25.9 MB 40 kB/s 
     |████████████████████████████████| 40 kB 7.7 MB/s 
     |████████████████████████████████| 72 kB 1.6 MB/s 
     |████████████████████████████████| 303 kB 72.4 MB/s 
     |████████████████████████████████| 80 kB 12.3 MB/s 
     |████████████████████████████████| 1.3 

In [ ]:
!nvcc --version
!nvidia-smi

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Sun_Jul_28_19:07:16_PDT_2019
Cuda compilation tools, release 10.1, V10.1.243
Tue Feb 16 13:32:06 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.39       Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   44C    P8     9W /  70W |      0MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+

## Set default CUDA version

Default CUDA version in Colab is 10.1, need to change to 10.0

In [ ]:
! echo $PATH

import os
os.environ['PATH'] += ":/usr/local/cuda-10.0/bin"
os.environ['CUDADIR'] = "/usr/local/cuda-10.0"
os.environ['LD_LIBRARY_PATH'] = "/usr/lib64-nvidia:/usr/local/cuda-10.0/lib64"

!echo $PATH
!echo $LD_LIBRARY_PATH
!source ~/.bashrc

/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin:/opt/bin
/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin:/opt/bin:/usr/local/cuda-10.0/bin
/usr/lib64-nvidia:/usr/local/cuda-10.0/lib64


In [ ]:
!env | grep -i cuda

LD_LIBRARY_PATH=/usr/lib64-nvidia:/usr/local/cuda-10.0/lib64
CUDADIR=/usr/local/cuda-10.0
LIBRARY_PATH=/usr/local/cuda/lib64/stubs
CUDA_PKG_VERSION=10-1=10.1.243-1
CUDA_VERSION=10.1.243
NVIDIA_REQUIRE_CUDA=cuda>=10.1 brand=tesla,driver>=396,driver<397 brand=tesla,driver>=410,driver<411 brand=tesla,driver>=418,driver<419
PATH=/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin:/opt/bin:/usr/local/cuda-10.0/bin


In [ ]:
%cd /content/
!wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-repo-ubuntu1804_10.0.130-1_amd64.deb
!sudo apt-get install freeglut3 freeglut3-dev libxi-dev libxmu-dev
!sudo apt-get install build-essential dkms
!sudo dpkg -i cuda-repo-ubuntu1804_10.0.130-1_amd64.deb
!sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/7fa2af80.pub

!sudo apt-get update
!sudo apt-get install cuda-10-0

/content
--2021-02-16 13:32:16--  https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-repo-ubuntu1804_10.0.130-1_amd64.deb
Resolving developer.download.nvidia.com (developer.download.nvidia.com)... 152.195.19.142
Connecting to developer.download.nvidia.com (developer.download.nvidia.com)|152.195.19.142|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2940 (2.9K) [application/x-deb]
Saving to: ‘cuda-repo-ubuntu1804_10.0.130-1_amd64.deb’

cuda-repo-ubuntu180 100%[===================>]   2.87K  --.-KB/s    in 0s      

2021-02-16 13:32:16 (158 MB/s) - ‘cuda-repo-ubuntu1804_10.0.130-1_amd64.deb’ saved [2940/2940]

Reading package lists... Done
Building dependency tree       
Reading state information... Done
libxi-dev is already the newest version (2:1.7.9-1).
libxi-dev set to manually installed.
libxmu-dev is already the newest version (2:1.1.2-2).
libxmu-dev set to manually installed.
freeglut3 is already the newest version (2.8.1-3).


In [ ]:
!sudo rm /usr/local/cuda
!sudo ln -s /usr/local/cuda-10.0 /usr/local/cuda
%ls -l /usr/local/

total 72
drwxr-xr-x  1 root root 4096 Feb 16 13:31 bin/
lrwxrwxrwx  1 root root   20 Feb 16 13:32 cuda -> /usr/local/cuda-10.0/
drwxr-xr-x 16 root root 4096 Feb 10 14:31 cuda-10.0/
drwxr-xr-x  1 root root 4096 Feb 10 14:33 cuda-10.1/
drwxr-xr-x  1 root root 4096 Feb 10 14:41 etc/
drwxr-xr-x  2 root root 4096 Nov 19 13:07 games/
drwxr-xr-x  2 root root 4096 Feb 10 14:51 _gcs_config_ops.so/
drwxr-xr-x  1 root root 4096 Feb 10 15:00 include/
drwxr-xr-x  1 root root 4096 Feb 10 15:00 lib/
-rw-r--r--  1 root root 1636 Feb 10 14:54 LICENSE.txt
drwxr-xr-x  3 root root 4096 Feb 10 14:51 licensing/
lrwxrwxrwx  1 root root    9 Nov 19 13:07 man -> share/man/
drwxr-xr-x  2 root root 4096 Nov 19 13:09 sbin/
-rw-r--r--  1 root root 7291 Feb 10 14:54 setup.cfg
drwxr-xr-x  1 root root 4096 Feb 10 14:41 share/
drwxr-xr-x  2 root root 4096 Nov 19 13:07 src/
drwxr-xr-x  2 root root 4096 Feb 10 15:01 xgboost/


In [ ]:
!pip3 uninstall tensorflow
!pip3 install 'tensorflow-gpu==1.15.2'

Found existing installation: tensorflow 1.15.2
Uninstalling tensorflow-1.15.2:
  Would remove:
    /usr/local/bin/estimator_ckpt_converter
    /usr/local/bin/freeze_graph
    /usr/local/bin/saved_model_cli
    /usr/local/bin/tensorboard
    /usr/local/bin/tf_upgrade_v2
    /usr/local/bin/tflite_convert
    /usr/local/bin/toco
    /usr/local/bin/toco_from_protos
    /usr/local/lib/python3.6/dist-packages/tensorflow-1.15.2.dist-info/*
    /usr/local/lib/python3.6/dist-packages/tensorflow/*
    /usr/local/lib/python3.6/dist-packages/tensorflow_core/*
Proceed (y/n)? y
  Successfully uninstalled tensorflow-1.15.2
     |████████████████████████████████| 411.0 MB 36 kB/s 


## If you get a broken-install error while installing, run this

In [ ]:
!sudo apt-get -o Dpkg::Options::="--force-overwrite" install --fix-broken

Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 31 not upgraded.


## Copy data to Colab from Google Bucket

In [ ]:
%cd /content/

/content


In [ ]:
!gdown --id 1NDwoMFAE0brBbXOECrqGueDBB1rS3lEp
!gdown --id 1qS65YOp5p8wUCEx-FFCRwAAuFOyrXZcX

!unzip hindi_female_english.zip -d /content/hindi_female_english
!unzip hindi_male_english.zip -d /content/hindi_male_english

Streaming output truncated to the last 5000 lines.
  inflating: /content/hindi_male_english/english/wav/engtext_2230.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_2426.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_1831.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_2341.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_2479.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_1243.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_819.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_233.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_327.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_1697.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_2833.wav  
  inflating: /content/hindi_male_english/english/wav/engtext_2666.wav  
  inflating: /content/hindi_male_english/english/wav/arctic_b0383.wav  
  inflating: /co

In [ ]:
!mv /content/hindi_female_english /content/hindi_female_english_org
!mv /content/hindi_male_english /content/hindi_male_english_org

!mkdir hindi_female_english
!mkdir hindi_female_english/english
!mkdir hindi_female_english/english/wav

!mkdir /content/hindi_male_english
!mkdir /content/hindi_male_english/english
!mkdir /content/hindi_male_english/english/wav

%cd /content/
!python3 sox_test.py

Streaming output truncated to the last 5000 lines.
engtext_3970.wav
engtext_3853.wav
engtext_1323.wav
arctic_a0318.wav
engtext_741.wav
engtext_1033.wav
engtext_1257.wav
arctic_b0114.wav
engtext_3843.wav
engtext_2963.wav
engtext_2331.wav
engtext_1881.wav
engtext_486.wav
arctic_b0280.wav
engtext_2100.wav
engtext_747.wav
engtext_1040.wav
engtext_3683.wav
engtext_3619.wav
engtext_1660.wav
engtext_3260.wav
engtext_2814.wav
engtext_3511.wav
arctic_a0159.wav
arctic_a0289.wav
arctic_b0515.wav
engtext_2568.wav
engtext_2696.wav
engtext_1255.wav
engtext_2772.wav
engtext_1578.wav
engtext_1435.wav
engtext_3656.wav
engtext_3132.wav
engtext_1271.wav
arctic_b0188.wav
engtext_2027.wav
engtext_427.wav
engtext_1999.wav
engtext_3607.wav
engtext_1277.wav
engtext_610.wav
engtext_2442.wav
arctic_b0139.wav
engtext_1513.wav
arctic_a0030.wav
engtext_2690.wav
engtext_863.wav
engtext_3202.wav
engtext_120.wav
engtext_3089.wav
engtext_1942.wav
arctic_a0264.wav
arctic_a0414.wav
engtext_3213.wav
engtext_2116.wav
engt

## Check for extra characters not in alphabet.txt

---

You might need to comment out some lines in `check_characters.py` to get it to work. Run this for all three files as a sanity check before training to avoid future errors.

NOTE: Don't change the original alphabet until you know exactly what you're doing. If you use the original alphabet, you can use the scorer file to improve predictions.

After commenting out some lines, this is how line 44 should look like
```
if not args.disable_unicode_variants:
  unicode_transcript = unicodedata.normalize("NFKC", row[2])
  #if row[2] != unicode_transcript:
    #print("Your input file", in_file, "contains at least one transript with unicode chars on more than one code-point: '{}'. Consider using NFKC normalization: unicodedata.normalize('NFKC', str).".format(row[2]))
    #sys.exit(-1)
all_text |= set(row[2])

```



In [ ]:
%cd /content/
!pip install num2words
!sudo rm /content/test.csv /content/train.csv /content/output.csv /content/dev.csv 
!python pre_process.py --wav /content/hindi_female_english/english/wav --meta /content/hindi_female_english/english


Streaming output truncated to the last 5000 lines.
arctic_a0132.wav 114274 a cry of joy burst from philip's lips
arctic_a0133.wav 153746 philip began to feel that he had foolishly overestimated his strength
arctic_a0134.wav 86330 he obeyed the pressure of her hand
arctic_a0135.wav 123380 i am going to surprise father and you will go with pierre
arctic_a0136.wav 160370 about him everywhere were the evidences of luxury and of age
arctic_a0137.wav 99940 then he stepped back with a low cry of pleasure
arctic_a0138.wav 142912 in the picture he saw each moment a greater resemblance to jeanne
arctic_a0139.wav 172766 he told himself that as he washed himself and groomed his disheveled clothes
arctic_a0140.wav 109366 accept a father's blessing and with it this
arctic_a0141.wav 109408 it seems like a strange pointing of the hand of god
arctic_a0142.wav 104960 such things had occurred before he told philip
arctic_a0143.wav 90852 ah i had forgotten he exclaimed
arctic_a0144.wav 146656 but there wa

In [ ]:
# Hash a line in pre_process.py


!python pre_process.py --wav /content/hindi_male_english/english/wav --meta /content/hindi_male_english/english


Streaming output truncated to the last 5000 lines.
arctic_a0136.wav 177222 about him everywhere were the evidences of luxury and of age
arctic_a0137.wav 108294 then he stepped back with a low cry of pleasure
arctic_a0138.wav 146440 in the picture he saw each moment a greater resemblance to jeanne
arctic_a0139.wav 171946 he told himself that he washed himself and groomed his disheveled clothes
arctic_a0140.wav 115006 accept a father's blessing and with it this
arctic_a0141.wav 118068 it seems like a strange pointing of the hand of god
arctic_a0142.wav 123216 such things had occurred before he told philip
arctic_a0143.wav 108652 ah i had forgotten he exclaimed
arctic_a0144.wav 140014 but there was something even more startling than this resemblance
arctic_a0145.wav 136994 i have to be careful of them as they tear very easily
arctic_a0146.wav 128244 of course that is uninteresting she continued
arctic_a0147.wav 173768 a moment before he was intoxicated by a joy that was almost madness
arc

In [ ]:
# Correct output.py before splitting

%cd /content/
!python split.py

/content


In [ ]:
!python3 /content/DeepSpeech/training/deepspeech_training/util/check_characters.py -csv /content/train.csv -alpha
!python3 /content/DeepSpeech/training/deepspeech_training/util/check_characters.py -csv /content/dev.csv -alpha
!python3 /content/DeepSpeech/training/deepspeech_training/util/check_characters.py -csv /content/test.csv -alpha

### Reading in the following transcript files: ###
### ['/content/train.csv'] ###
### The following unique characters were found in your transcripts: ###
x
g
f
o
m
'
w
j
r
z
 
n
h
c
d
p
e
u
i
l
q
v
t
b
s
-
y
k
a
### ^^^ You can copy-paste these into data/alphabet.txt ###
### Reading in the following transcript files: ###
### ['/content/dev.csv'] ###
### The following unique characters were found in your transcripts: ###
s
'
h
 
t
g
d
p
o
q
l
w
u
x
b
n
m
a
r
f
i
e
y
j
c
v
k
z
### ^^^ You can copy-paste these into data/alphabet.txt ###
### Reading in the following transcript files: ###
### ['/content/test.csv'] ###
### The following unique characters were found in your transcripts: ###
t
g
k
d
l
h
s
r
v
q
n
'
e
o
w
a
z
y
i
f
p
u
 
j
c
x
m
b
### ^^^ You can copy-paste these into data/alphabet.txt ###


## Training

In [ ]:
%cd /content/

!mkdir models

%cd /content/models/

!wget https://github.com/mozilla/DeepSpeech/releases/download/v0.7.4/deepspeech-0.7.4-models.pbmm
!wget https://github.com/mozilla/DeepSpeech/releases/download/v0.7.4/deepspeech-0.7.4-models.scorer

%cd /content/
!mkdir model_checkpoints/
!tar -xvf deepspeech-0.7.4-checkpoint.tar.gz -C /content/model_checkpoints/ 


/content
/content
mkdir: cannot create directory ‘model_checkpoints/’: File exists
._deepspeech-0.7.4-checkpoint
deepspeech-0.7.4-checkpoint/
deepspeech-0.7.4-checkpoint/._checkpoint
deepspeech-0.7.4-checkpoint/checkpoint
deepspeech-0.7.4-checkpoint/._best_dev-732522.data-00000-of-00001
deepspeech-0.7.4-checkpoint/best_dev-732522.data-00000-of-00001
^C


In [ ]:
%cd /content/DeepSpeech
!python3 DeepSpeech.py --train_cudnn True --early_stop True --es_epochs 5 --n_hidden 2048 --epochs 30 \
  --scorer_path /content/models/deepspeech-0.7.4-models.scorer \
  --export_dir /content/models/ --checkpoint_dir /content/model_checkpoints/deepspeech-0.7.4-checkpoint/ \
  --train_files /content/train.csv --dev_files /content/dev.csv --test_files /content/test.csv \
  --learning_rate 0.001 --train_batch_size 64 --test_batch_size 32 --dev_batch_size 32 --export_file_name 'ft_model' \
  --augment reverb[p=0.2,delay=50.0~30.0,decay=10.0:2.0~1.0] \
  --augment volume[p=0.2,dbfs=-10:-40] \
  --augment pitch[p=0.2,pitch=1~0.2] \
  --augment tempo[p=0.2,factor=1~0.5] 

/content/DeepSpeech
I0216 19:36:12.495881 139656099903360 utils.py:157] NumExpr defaulting to 2 threads.
I Loading best validating checkpoint from /content/model_checkpoints/deepspeech-0.7.4-checkpoint/best_dev-735846
I Loading variable from checkpoint: beta1_power
I Loading variable from checkpoint: beta2_power
I Loading variable from checkpoint: cudnn_lstm/opaque_kernel
I Loading variable from checkpoint: cudnn_lstm/opaque_kernel/Adam
I Loading variable from checkpoint: cudnn_lstm/opaque_kernel/Adam_1
I Loading variable from checkpoint: global_step
I Loading variable from checkpoint: layer_1/bias
I Loading variable from checkpoint: layer_1/bias/Adam
I Loading variable from checkpoint: layer_1/bias/Adam_1
I Loading variable from checkpoint: layer_1/weights
I Loading variable from checkpoint: layer_1/weights/Adam
I Loading variable from checkpoint: layer_1/weights/Adam_1
I Loading variable from checkpoint: layer_2/bias
I Loading variable from checkpoint: layer_2/bias/Adam
I Loading var

## Exporting the model for inference

In [ ]:
%cd /content/DeepSpeech/
!python3 util/taskcluster.py --source tensorflow --artifact convert_graphdef_memmapped_format --branch r1.15 --target .
!./convert_graphdef_memmapped_format --in_graph=/content/models/ft_model.pb --out_graph=/content/models/male_female_mix_high_LR.pbmm

/content/DeepSpeech
File already exists: /content/DeepSpeech/convert_graphdef_memmapped_format
2021-02-16 20:08:16.174726: I tensorflow/contrib/util/convert_graphdef_memmapped_format_lib.cc:171] Converted 7 nodes
